File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/SaveOptOutForCCPABridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize
import com.usercentrics.sdk.models.settings.UsercentricsConsentType

internal class SaveOptOutForCCPABridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "saveOptOutForCCPA"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        val argsMap = call.arguments as Map<*, *>
        val consents = usercentrics.instance.saveOptOutForCCPA(
            isOptedOut = argsMap["isOptedOut"] as Boolean,
            consentType = UsercentricsConsentType.valueOf(argsMap["consentType"] as String)
        )
        result.success(consents.map { it.serialize() })
    }

}

File: ios/Classes/Bridge/SaveOptOutForCCPABridge.swift
import Usercentrics

struct SaveOptOutForCCPABridge : MethodBridge {

    let name: String = "saveOptOutForCCPA"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let isOptedOut = argsDict["isOptedOut"] as! Bool
        let consentType = UsercentricsConsentType.initialize(from: argsDict["consentType"])!
        let consents = usercentrics.shared.saveOptOutForCCPA(isOptedOut: isOptedOut, consentType: consentType)
        result(consents.map { $0.serialize() })
    }
}
